File: examples/cpp/map_example.cpp
/**
 * Copyright 2025 Bang-Shien Chen.
 * All rights reserved. See LICENSE for the license information.
 */

#include <iostream>

#include "problem.h"
#include "solver.h"

int main() {
  size_t max_iter = 1000;
  double tol = 1e-6;
  double rel_tol = 1e-6;
  bool verbose = true;

  int m = 10;
  int n = 10;

  MAP::Problem problem;
  auto [mat_A, mat_B, vec_c, gtruth] = problem.generate_problem1(m, n);

  MAP::Solver solver(max_iter, tol, rel_tol, verbose);
  Eigen::VectorXd sol = solver.solve(mat_A, mat_B, vec_c);

  std::cout << "Ground Truth:" << "\n" << gtruth.transpose() << "\n\n";
  std::cout << "MAP solution:" << "\n" << sol.transpose() << "\n\n";

  return 0;
}

File: examples/python/map_example.py
# Copyright 2025 Bang-Shien Chen.
# All rights reserved. See LICENSE for the license information.

try:
    import map_python
except ImportError:
    print("Make sure project was built with python binding.")
    exit(1)


def main():
    max_iter = 1000
    tol = 1e-6
    rel_tol = 1e-6

    m = 10
    n = 10

    problem = map_python.Problem()
    mat_A, mat_B, vec_c, gtruth = problem.generate_problem1(m, n)

    solver = map_python.Solver(
        max_iteration=max_iter, tolerance=tol, relative_tolerance=rel_tol, verbose=True
    )
    sol = solver.solve(mat_A, mat_B, vec_c)

    print("Ground truth:")
    print(gtruth, end="\n\n")

    print("MAP solution:")
    print(sol, end="\n\n")


if __name__ == "__main__":
    main()
